lib/normalizer.c

```diff
@@ -1418,6 +1418,89 @@ grn_nfkc_normalize_unify_katakana_v_sounds(grn_ctx *ctx,
   return current;
 }
 
+static const unsigned char *
+grn_nfkc_normalize_unify_katakana_z_sounds(grn_ctx *ctx,
+                                           const unsigned char *start,
+                                           const unsigned char *current,
+                                           const unsigned char *end,
+                                           size_t *n_used_bytes,
+                                           size_t *n_used_characters,
+                                           unsigned char *unified_buffer,
+                                           size_t *n_unified_bytes,
+                                           size_t *n_unified_characters,
+                                           void *user_data)
+{
+  size_t char_length;
+  bool start_char;
+
+  char_length = (size_t)grn_charlen_(ctx, current, end, GRN_ENC_UTF8);
+  start_char = false;
+
+  *n_used_bytes = char_length;
+  *n_used_characters = 1;
+
+  if (char_length == 3) {
+    /* U+30C5 KATAKANA LETTER DU */
+    /* U+30BA KATAKANA LETTER ZU */
+    start_char = 
+      (current[0] == 0xe3 && current[1] == 0x83 && current[2] == 0x85) ||
+      (current[0] == 0xe3 && current[1] == 0x82 && current[2] == 0xba);
+  }
+
+  if (start_char) {
```

Comment:
If you want to introduce a descriptive variable for this condition, could you use more meaningful name like `is_du_or_zu` or something?

In this case, I don't thing that we need a descriptive variable because the following code is well descriptive but it's not a strong opinion:

```c
if ((char_length == 3) &&
    /* U+30C5 KATAKANA LETTER DU */
    (current[0] == 0xe3 && current[1] == 0x83 && current[2] == 0x85) ||
    /* U+30BA KATAKANA LETTER ZU */
    (current[0] == 0xe3 && current[1] == 0x82 && current[2] == 0xba)) {
```

Reply:
We need to add extra parentheses because `&&` has higher priority than `||`, and I feel it somehow undermines readability.
That is the reason why I introduced a descriptive variable. 

```
if ((char_length == 3) &&
    /* U+30C5 KATAKANA LETTER DU */
    ((current[0] == 0xe3 && current[1] == 0x83 && current[2] == 0x85) ||
    /* U+30BA KATAKANA LETTER ZU */
    (current[0] == 0xe3 && current[1] == 0x82 && current[2] == 0xba))) 
```

But this is also not a strong opinion...
As a conclusion, I will remove the descriptive variable and fix it like above.

Reply:
Fixed